app/battle/battle.py

```diff
@@ -0,0 +1,15 @@
+def knights_battle(knight1: dict, knight2: dict) -> dict:
+    knight1["hp"] -= knight2["power"] - knight1["protection"]
+    knight2["hp"] -= knight1["power"] - knight2["protection"]
```

Comment:
The current formula for calculating the damage (`knight1['hp'] -= knight2['power'] - knight1['protection']`) does not account for the possibility of negative damage. If `knight1['protection']` is greater than `knight2['power']`, this will increase `knight1['hp']` instead of decreasing it. Consider using `max(0, knight2['power'] - knight1['protection'])` to ensure that the damage is not negative.